inc/commontreedropdown.class.php

```diff
@@ -188,8 +189,8 @@ function prepareInputForUpdate($input) {
             $input["ancestors_cache"] = '';
             if (Toolbox::useCache()) {
                $ckey = $this->getTable() . '_ancestors_cache_' . $this->getID();
-               if (apcu_exists($ckey)) {
-                  apcu_delete($ckey);
+               if ($GLPI_CACHE->hasItem($ckey)) {
+                  $GLPI_CACHE->removeItem($ckey);
                }
```

Comment:
but GPI_CACHE may be "false" ?

Reply:
This is handle from Toolbox::useCache()

Reply:
Te make things clear: the `Toolbox::useCache()` will return false if `$GLPI_CACHE` is false. The only exception is from tests, when the `CACHED_TESTS` constant is defined (that's why I had to declare `$GLPI_CACHE` in the tests).

Reply:
ok